meteor/client/ui/SegmentTimeline/Renderers/VTSourceRenderer.tsx

```diff
@@ -303,70 +307,50 @@ export class VTSourceRendererBase extends CustomLayerItemRenderer<IProps & WithT
 
 	getScenes = (): Array<number> | undefined => {
 		if (this.props.piece) {
-			const itemDuration = this.getItemDuration() // todo: rename to pieceDuration
-
 			const piece = this.props.piece
 			if (piece.contentPackageInfos) {
 				// TODO: support multiple packages:
 				const contentPackageInfos = Object.values(piece.contentPackageInfos)
 				if (contentPackageInfos[0]?.deepScan?.scenes) {
-					const scenes = _.compact(
-						contentPackageInfos[0].deepScan.scenes.map((i) => {
-							const sceneTime = i * 1000
-							if (sceneTime < itemDuration) {
```

Comment:
Doesn't this capping of `sceneTime` at `itemDuration` serve some purpose? I'm almost sure this filter here was a result of a bug somewhere.

Reply:
I had doubts on this last week as well.
This is actually a bug-fix; IIRC @ianshade explained that this is fixing a situation where scenes/blacks/freezes was filtered for only the original duration of a Piece (like when it was capped in a Part). If the used decided to play the Part longer than planned, the scenes/blacks/freezes wouldn't show up for the lengthened part of the Piece.

The scenes are [ultimately filtered later](https://github.com/nrkno/sofie-core/blob/sync/release41/meteor/client/ui/SegmentTimeline/Renderers/VTSourceRenderer.tsx#L546), so this should be OK.

With this being said, I've got this on my list of Things-to-test, to ensure that it still works properly.

Reply:
That's a good find, I totally forgot that they are filtered later! Yeah, if they are filtered later to make sure we don't produce more DOM elements than we need, then this definitely makes sense.